juju/juju
# feat: remove upgrade series/base

This changeset removes all of the upgrade series/base work. We're not going to transition this work over to 4.0, so we can just remove it. This may be put back in the future, but it won't be done for the initial release of 4.0.

The additional benefit to the removal of the work is that it does simpify the machine and charm epics.

Work to ensure that 3.6 agent doesn't crash when faced with missing charms will also need to be done. That work will be done against the 3.6 branch.





## Checklist




## QA steps

Migration should succeed:

```sh
$ juju bootstrap lxd src
$ juju add-model moveme
$ juju deploy ubuntu
$ juju bootstrap lxd dst
$ juju migrate src:moveme dst
```

Upgrade base should no longer be present:

```sh
$ juju upgrade-base
ERROR juju: "upgrade-base" is not a juju command. See "juju --help".

Did you mean:
        upgrade-model
```


## Documentation changes

@tmihoc Removal of update-base/series along with their hooks.

## Links



**Jira card:** [JUJU-6330](https://warthogs.atlassian.net/browse/JUJU-6330)


[JUJU-6330]: https://warthogs.atlassian.net/browse/JUJU-6330?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
